go/vt/vttablet/tabletmanager/shard_sync.go

```diff
@@ -187,25 +189,34 @@ func syncShardMaster(ctx context.Context, ts *topo.Server, tablet *topodatapb.Ta
 // If active reparents are disabled, we don't touch our MySQL.
 // We just directly update our tablet type to REPLICA.
 func (agent *ActionAgent) abortMasterTerm(ctx context.Context, masterAlias *topodatapb.TabletAlias) error {
```

Comment:
at some point we should rename this to `endMasterTerm`.

Reply:
The name was meant to convey that this is not the "graceful" way for a master term to come to an end. If we hit this code path, it means we should sort of "force demote" ourselves because another master has already taken over behind our backs. Do you think `abort` is a bad way to describe this?

Reply:
That sounds right, but it means that we are ending a master term in a not "graceful" way with the current implementation of PRS. Are we ok with that?

Reply:
PRS should have already called DemoteMaster to gracefully step down before we get to this code path. We will still technically execute this code path, but once we make these idempotent, there will be no net effect from this code path in the case of a successful PRS.